pkg/server/handlers.go

```diff
@@ -30,6 +35,64 @@ func HostsHandler(w http.ResponseWriter, r *http.Request) {
 	RespondWithJson(w, hostList)
 }
 
+type InstallConfig struct {
+	ClusterName string `json:"clusterName"`
+	DNSDomain   string `json:"DNSDomain"`
+	PullSecret  string `json:"pullSecret"`
+}
+
+type DefineClusterCredentials struct {
+	ClusterName string `json:"clusterName"`
+	DNSDomain   string `json:"DNSDomain"`
+	Username    string `json:"username"`
+	Password    string `json:"password"`
+}
+
+func CreateInstallConfig(w http.ResponseWriter, r *http.Request) {
+	var clusterDefinition DefineClusterCredentials
+	_ = json.NewDecoder(r.Body).Decode(&clusterDefinition)
+
+	tokenRequestURL := fmt.Sprintf("https://developers.redhat.com/auth/realms/rhd/protocol/openid-connect/token")
+
+	resp, err := http.PostForm(tokenRequestURL, url.Values{
+		"grant_type": {"password"},
+		"client_id":  {"uhc"},
+		"username":   {clusterDefinition.Username},
+		"password":   {clusterDefinition.Password},
+	})
+
+	if err != nil {
+		log.Print("Error in api call", err)
```

Comment:
After logging the error, we should return an error to the client. Also, later code will fail as the `resp` variable is unpopulated. 

```go
RespondWithError(w, err, 500)
return
```